specification/search/data-plane/Azure.Search/preview/2025-08-01-preview/searchservice.json

```diff
@@ -0,0 +1,14691 @@
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Statistics summary for all indexes.",
+            "schema": {
+              "$ref": "#/definitions/ListIndexStatsSummary"
+            }
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/data-plane/v1/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    }
+  },
+  "definitions": {
+    "KnowledgeAgent": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the knowledge agent."
+        },
+        "models": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeAgentModel"
+          },
+          "description": "Contains configuration options on how to connect to AI models."
+        },
+        "knowledgeSources": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeSourceReference"
+          }
+        },
+        "outputConfiguration": {
+          "$ref": "#/definitions/KnowledgeAgentOutputConfiguration"
+        },
+        "requestLimits": {
+          "$ref": "#/definitions/KnowledgeAgentRequestLimits"
+        },
+        "retrievalInstructions": {
+          "type": "string",
+          "description": "Instructions considered by the knowledge agent when developing query plan."
+        },
+        "@odata.etag": {
+          "x-ms-client-name": "ETag",
+          "type": "string",
+          "description": "The ETag of the agent."
+        },
+        "encryptionKey": {
+          "$ref": "#/definitions/SearchResourceEncryptionKey",
+          "description": "A description of an encryption key that you create in Azure Key Vault. This key is used to provide an additional level of encryption-at-rest for your agent definition when you want full assurance that no one, not even Microsoft, can decrypt them. Once you have encrypted your agent definition, it will always remain encrypted. The search service will ignore attempts to set this property to null. You can change this property as needed if you want to rotate your encryption key; Your agent definition will be unaffected. Encryption with customer-managed keys is not available for free search services, and is only available for paid services created on or after January 1, 2019.",
+          "externalDocs": {
+            "url": "https://aka.ms/azure-search-encryption-with-cmk"
+          },
+          "x-nullable": true
+        },
+        "description": {
+          "type": "string",
+          "description": "The description of the agent."
+        }
+      },
+      "required": [
+        "name",
+        "models"
```

Comment:
Should `knowledgeSources` be required? What happens when a Knowledge Agent is created without sources?